source/hwIo.py

```diff
@@ -178,9 +177,31 @@ def close(self):
 
 	def _notifyReceive(self, data):
 		# Set the timeout for onReceive in case it does a sync read.
-		self._ser.timeout = self._origTimeout
+		self._setTimeout(self._origTimeout)
 		super(Serial, self)._notifyReceive(data)
-		self._ser.timeout = None
+		self._setTimeout(None)
+
+	def _setTimeout(self, timeout):
+		# #6035: pyserial reconfigures all settings of the port when setting a timeout.
+		# This can cause error 'Cannot configure port, some setting was wrong.'
+		# Therefore, manually set the timeouts using the Win32 API.
+		# Adapted from pyserial 3.1.1.
+		timeouts = COMMTIMEOUTS()
+		if timeout is None:
+			pass # default of all zeros is OK

```

Comment:
I'd prefer we didn't have empty blocks in an `if` like this. I'd change the `else` below to `elif timeout is not None`. Same with the later `pass` statements.